### PR TITLE
Translate-c self-hosted var decl and remove translate mode

### LINKED ISSUE #2780: translate-c: remove concept of "translate mode"
(Please feel free to add a more thorough description)

From [conversations on IRC](https://freenode.irclog.whitequark.org/zig/2019-06-23):

```
20:25 <hryx> what's the TLDR difference between include mode and -c mode?
20:26 <andrewrk> the idea was that in -c mode, you're translating a C project, so non-static functions should be `export` because that's what -c would do
20:26 <andrewrk> but in #include mode, it's like you just want to call the functions
20:26 <andrewrk> now that I'm thinking about it again though, I don't think there should be a difference
20:27 <andrewrk> because if you were to #include a non-static function in C, it would get exported. so the @cImport equivalent of that would be indeed, creating an export function and translating its definition
20:27 <andrewrk> so thinking about it again, I don't see why there needs to be 2 modes
```

This conversation resulted from work on #2569 .

### PR DESCRIPTION
**Pull request by Vexu**
Closes  #2780

### CODE REVIEW

File: src-self-hosted/translate_c.zig [hunk 1]
```diff
@@ -310,6 +302,59 @@ fn visitFnDecl(c: *Context, fn_decl: *const ZigClangFunctionDecl) Error!void {
...
+        },
+        error.OutOfMemory => |e| return e,
+    };
+
+    if (is_static and !is_extern) {
+        const eq_tok = try appendToken(c, .Equal, "=");
+        const init_node = if (ZigClangVarDecl_hasInit(var_decl)) blk: {
+            const ap_value = ZigClangVarDecl_evaluateValue(var_decl) orelse
+                return failDecl(c, var_decl_loc, var_name, "unable to evaluate initializer", .{});
+            break :blk transApValue(rp, ap_value, qual_type, var_decl_loc) catch |err| switch (err) {
```
Comment (Vexu) [inline src-self-hosted/translate_c.zig]: This should probably just use `ZigClangVarDecl_getInit` and then use`transExpr`.

**Comment by Vexu** [comment]
Removing translate mode from stage 1 exposed some bugs in translate-c so I reverted it because I would rather just finish translate-c-2.

**Comment by andrewrk** [comment]
I'm really excited to see this progress on self-hosted translate-c! :rocket:

**Comment by kavika13** [comment]
@Vexu are you in discord/IRC? I also plan to bang on this a bit too, want to not dupe effort if I can help it.
If not, no worries. My progress is going to be slow anyhow :)

**Pull request merged**